pageserver/src/tenant.rs

```diff
@@ -3767,13 +3767,17 @@ async fn run_initdb(
         .spawn()?;
 
     tokio::select! {
-        initdb_output = initdb_command.wait_with_output() => {
-            let initdb_output = initdb_output?;
-            if !initdb_output.status.success() {
-                return Err(InitdbError::Failed(initdb_output.status, initdb_output.stderr));
+        exit_status = initdb_command.wait() => {
+            let exit_status = exit_status?;
+            if !exit_status.success() {
+                let mut stderr = initdb_command.stderr.take().unwrap();
+                let mut stderr_vec = Vec::new();
+                tokio::io::copy(&mut stderr, &mut stderr_vec).await?;
+                return Err(InitdbError::Failed(exit_status, stderr_vec));
             }
         }
         _ = cancel.cancelled() => {
+            initdb_command.kill().await?;
```

Comment:
Shouldn't this arm still wait for the process to exit?

Reply:
Oh you mean the `kill()` function, despite being `async`, actually never awaits the exit of the process? That might be it, trying...

Reply:
I tried it, it didn't fix the issue.